Export XKB_DEFAULT_RULES=evdev when using Wayland or GBM to avoid
surprises and make sure all keys work.  Default rules are set to
'xorg' in x11/libxkbcommon which is problematic.

--- tools/Linux/kodi.sh.in.orig	2019-01-17 22:53:57 UTC
+++ tools/Linux/kodi.sh.in
@@ -83,6 +83,11 @@ if [ ! -x ${KODI_BINARY} ]; then
     exit 2
 fi
 
+if [ "${KODI_BINARY##*-}" == "gbm" ] || [ "${KODI_BINARY##*-}" == "wayland" ]; then
+    : ${XKB_DEFAULT_RULES:=evdev}
+    export XKB_DEFAULT_RULES
+fi
+
 APPORT_CORE="/var/crash/$(echo -n ${KODI_BINARY}|tr / _).$(id -u).crash"
 
 migrate_home()
